File: test_commits/javascript/bug_fix_changed3.js
async function fetchUserData(userId) {
  try {
      const response = await fetch(`/api/user/${userId}`);
      if (!response.ok) {
          throw new Error("Network response was not ok");
      }
      return await response.json();
  } catch (error) {
      console.error("Failed to fetch user data:", error);
      return null;
  }
}

File: test_commits/python/bug_fix_changed3.py
import requests

def get_weather(city):
    try:
        response = requests.get(f"http://api.weather.com/{city}")
        response.raise_for_status()
        return response.json()
    except requests.RequestException as e:
        print(f"Error fetching weather data: {e}")
        return None
